core/src/main/java/org/elasticsearch/index/query/AndQueryBuilder.java

```diff
@@ -110,13 +110,19 @@ public Query toQuery(QueryParseContext parseContext) throws QueryParsingExceptio
         }
 
         BooleanQuery query = new BooleanQuery();
+        int addedFilters = 0;
         for (QueryBuilder f : filters) {
             Query innerQuery = f.toQuery(parseContext);
             // ignore queries that are null
             if (innerQuery != null) {
                 query.add(innerQuery, Occur.MUST);
+                addedFilters++;
             }
         }
+        if (addedFilters == 0) {
+            // no inner lucene query exists, ignore upstream
+            return null;
```

Comment:
ah ok I think I get it, we wouldn't get here before cause what is now the empty_query was null and you could never call toQuery against it. Now we have the empty query that returns null and we have to move some null checks to the different toQuery methods.